skypy/galaxy/size.py

```diff
@@ -0,0 +1,214 @@
+        Otherwise, np.broadcast(mean, sigma).size samples are drawn.
+
+    Results
+    -------
+    physical_size : numpy.ndarray or astropy.Quantity
+        Physical distance for a given galaxy with a given magnitude, in [kpc].
+        If size is None and magnitude is a scalar, a single sample is returned.
+        If size is different from None and magnitude is scalar,
+        shape is (size,). If magnitude has shape (nm,) and size=None,
+        shape is (nm,).
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skypy.galaxy import size
+    >>> np.random.seed(12345)
+    >>> magnitude = -16.0
+    >>> alpha, beta, gamma, M0 = 0.21, 0.53, -1.31, -20.52
+    >>> sigma1, sigma2 = 0.48, 0.25
+    >>> size.late_type_lognormal(magnitude, alpha, beta, gamma, M0,\
+                                 sigma1, sigma2)
+    <Quantity 0.9850926 kpc>
+
+
+    References
+    ----------
+    ..[1] S. Shen, H.J. Mo, S.D.M. White, M.R. Blanton, G. Kauffmann, W. Voges,
+        J. Brinkmann, I. Csabai, Mon. Not. Roy. Astron. Soc. 343, 978 (2003).
+    '''
+
+    if size is None and np.shape(magnitude):
+        size = np.shape(magnitude)
+
+    r_bar = np.power(10, -0.4 * alpha * magnitude + (beta - alpha) *
+                     np.log10(1 + np.power(10, -0.4 * (magnitude - M0)))
+                     + gamma) * units.kpc
+
+    sigma_lnR = sigma2 + (sigma1 - sigma2) /\
+                         (1.0 + np.power(10, -0.8 * (magnitude - M0)))
+
+    return r_bar * np.random.lognormal(sigma=sigma_lnR, size=size)
+
+
+def early_type_lognormal(magnitude, a, b, M0, sigma1, sigma2, size=None):
+    '''Lognormal distribution for early-type galaxies.
+    This function provides a lognormal distribution for the physical size of
+    early-type galaxies, described by equations 12, 14 and 16 in [1].
+
+    Parameters
+    ----------
+    magnitude : float or array_like.
+        Galaxy magnitude at which evaluate the lognormal distribution.
+    a_mu, b_mu : float
+        Linear model parameters describing the mean size of galaxies,
+        equation 3.14 in [1].
+    sigma: float
+        Standard deviation of the lognormal distribution for the
+        physical radius of galaxies.
+    size : int or tuple of ints, optional.
+        Output shape. If the given shape is, e.g., (m, n, k),
+        then m * n * k samples are drawn. If size is None (default),
+        a single value is returned if mean and sigma are both scalars.
+        Otherwise, np.broadcast(mean, sigma).size samples are drawn.
+
+    Results
+    -------
+    physical_size : ndarray or astropy.Quantity
+        Physical distance for a given galaxy with a given magnitude, in [kpc].
+        If size is None and magnitude is a scalar, a single sample is returned.
+        If size is different from None and magnitude is scalar,
+        shape is (size,). If magnitude has shape (nm,) and size=None,
+        shape is (nm,).
```

Comment:
Should this be written in doc for `size` instead of `(m,n,k)`...?

Reply:
Not sure about this. This refers solely to the shape of the output.

Reply:
Ja exactly. But `size` is the output shape. So it might be confusing.